pkg/pgmodel/querier/series_set.go

```diff
@@ -25,20 +24,37 @@ const (
 
 // pgxSeriesSet implements storage.SeriesSet.
 type pgxSeriesSet struct {
-	rowIdx  int
-	rows    []timescaleRow
-	err     error
-	querier labelQuerier
+	rowIdx     int
+	rows       []timescaleRow
+	labelIDMap map[int64]labels.Label
+	err        error
+	querier    labelQuerier
 }
 
 // pgxSeriesSet must implement storage.SeriesSet
 var _ storage.SeriesSet = (*pgxSeriesSet)(nil)
 
 func buildSeriesSet(rows []timescaleRow, querier labelQuerier) storage.SeriesSet {
+	labelIDMap := make(map[int64]labels.Label)
+	for i := range rows {
```

Comment:
We should really have a `initializeLabelIDMap()` as said above, and that will reduce the redundancy here.